### PR TITLE
Use tsup and tsup-preset-solid for building packages

### PR DESCRIPTION
**Pull request by thetarnav**
Still WIP - wanted to trigger some CIs.
I still need to do the submodules - should I loop all of the directories in /core ever to get the entries?
Also, I've made the packages ESM by default ("type": "module") (I kinda require that in tsup-preset-solid, cjs is only shipped additionally for the older tooling (jest))

### CODE REVIEW

**Comment by thetarnav** [comment]
I cannot do anything about the way the chunks are made, which was my only issue with this setup.
Basically, the build generates ~100 files in /dist/chunks which contains all the shared logic - which in this case is ALL the logic.
That + additional ~30 dts chunks in /dist.
Those chunks are created because there are multiple entries, and esbuild wants to separate shared code so that you only import what's needed.
If this is an issue, then you could always just drop the additional entries and only use the main index one, then there won't be any additional chunks.

image

File: apps/docs/src/components/prose.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { clsx } from "clsx";
 import { splitProps } from "solid-js";
 import { Dynamic } from "solid-js/web";
 
-export const Prose = createPolymorphicComponent<"div">(props => {
+export const Prose = /*#__PURE__*/ createPolymorphicComponent<"div">(props => {
```
[the remark this answers is not in the document]
Comment (thetarnav) [reply apps/docs/src/components/prose.tsx:6]: This caused a lot of unrelated formatting changes though...

**Comment by fabien-ml** [comment]
> I cannot do anything about the way the chunks are made, which was my only issue with this setup. Basically, the build generates ~100 files in /dist/chunks which contains all the shared logic - which in this case is ALL the logic. That + additional ~30 dts chunks in /dist. Those chunks are created because there are multiple entries, and esbuild wants to separate shared code so that you only import what's needed. If this is an issue, then you could always just drop the additional entries and only use the main index one, then there won't be any additional chunks.
> 
> image

Will it be simpler to go the "separate package" way (built with tsup-preset-solid) ?

**Comment by thetarnav** [comment]
It would eliminate chunks but preserve the "you only import what's needed" benefit.
Also, it would be less to download if one needs only one component.
But it makes your life a little bit harder.
You can always have a `build --watch` running so that you can make changes in different packages and have it reflect, but things like "go to definition" will navigate you to /dist.
Not sure if I would make a separate package out of every component if I were you. Because that's a lot of components, that keep getting added. Splitting packages is good if there are clearly different modules.
Tree-shaking should work the same regardless of the splitting. It's only about dx.

**Comment by fabien-ml** [comment]
Like we've said on Discord, i've moved to `tsup-preset-solid`. 
When the lib become stable I will release a `v1.0.0` of each component in their own package and `@kobalte/core` will be the package that include all.

**Pull request closed without merging**